peer/bitfield_test.go

```diff
@@ -0,0 +1,103 @@
+package peer
+
+import (
+	"math/rand"
+	"testing"
+)
+
+var ntests int = 1000
+
+func TestGet(t *testing.T) {
+	bitfield := bitfield{0b11001100, 0b10101010}
+	expected := []bool{true, true, false, false, true, true, false, false, true, false, true, false, true, false, true, false}
+	for index, exp := range expected {
+		assertGet(t, exp, bitfield, index)
+	}
+}
+
+func TestGetRandomised(t *testing.T) {
+	for i := 0; i < ntests; i++ {
+		bf := generateBitfield(t)
+		var expected []bool
+
+		for _, byte := range bf {
+			for j := 7; j >= 0; j-- {
+				bit := (byte & (1 << j)) == (1 << j)
```

Comment:
Should `byte & (1 << j)` not be enough?

Reply:
Nevermind, it is not a boolean... `bit := byte&(1<<j) != 0` might be cleaner?

Reply:
yep, and might generate better code. will do.